### DISCUSSION TITLE
Adding .devcontainer

### DISCUSSION THREAD

**Discussion by asherwin86** [NiclasvanEyk/initializer-for-laravel#292] [Ideas]
Would be nice if there was a checkbox just to add the .devcontainer.

In my use case to fix the problem I had to:
 - launch sail
 - run ./vendor/bin/sail artisan sail:install --devcontainer
 - make modifications to the devcontainer.json
 - stop sail
 - launch with devcontainer.

1 checkbox to remove 3 of these steps would be a nice addition.

Thanks.

**Comment by NiclasvanEyk** [comment]
What a great great suggestion! I haven't used devcontainers that much (yet), but I really like the idea!

I'd add it as a checkbox in the "Devtools" section. 

Do you know if this could also be somehow integrated into the initialization process? E.g. as in "just open the project in the devcontainer, which then automatically runs `./initialize` if the script is still present"?

**Reply by asherwin86** [reply]
I dont think so no. The process of just running ./initialise is fine. If you can think of a way to do it using postCreateCommand would be what you are wanting. `"postCreateCommand": "if [ -f /path/to/initialize ]; then /path/to/initialize; fi"`
If you wanted a suggestion for improving that part it would be something like copying what composer does so you dont have to download a zip in your browser and unzip it. Something like `wget https://initializer.com/project.tgz && tar xzf project.tgz && cd project && bash initialize`

**Comment by NiclasvanEyk** [comment]
Also what modifications do you typically do to the devcontainers.json file and are they useful for others?

**Reply by asherwin86** [reply]
Just having whatever sail does is best. The only changes generally made are adding personalised extensions.

**Comment by NiclasvanEyk** [comment]
<img width="1163" alt="Screenshot 2023-05-13 at 17 05 47" src="https://github.com/NiclasvanEyk/initializer-for-laravel/assets/10284694/54aa6f1d-fe81-4cf9-80e4-e2d332f7fc37">

Just deployed a new version including [a checkbox for devcontainers](https://laravel.initializer.dev/#development-tools).

**Reply by asherwin86** [reply]
Awesome, thanks.
